### PR TITLE
Pi USB connection

### PR DESCRIPTION
**Pull request by jaredgreene1**
## overview

CURRENT DRAFT: Of course any comments are welcome, but I'm not yet requesting review on this - just starting to draft the PR. 

Enables usb connectivity with the pi (a bug remains - will elaborate)

This PR includes:

- [X] Chore work
- [ ] Bug fixes
- [ ] Backwards-compatible feature additions
- [ ] Breaking changes
- [ ] Documentation updates
- [ ] Test updates

## changelog

USB connectivity
Right now this sets up a DHCP server on a static ip over ethernet. Once big issue, which I'm working on solving now, is that this only work once. After disconnecting, an ip will not get assigned on reconnection unless a static ip is set for the interface in the resin network config. This is work that still needs to happen

- (Chore) Change description
- (Fix) Change description
- (Feature) Change description
- (Refactor) Change description
- (Docs) Change description
- (Tests) Change description

**Note**: If your PR incorporates many changes, or many different types of changes, please consider splitting it up into smaller PRs.

## review requests

Describe any specific requests for your reviewers here. Tag any people or groups as necessary.

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -7,20 +7,26 @@ FROM resin/raspberrypi3-python:3.5-slim-20170705
 
 WORKDIR /usr/src
 
-COPY ./api/requirements.txt ./requirements.txt 
 RUN apt-get update
-RUN apt-get install gcc make git g++
+RUN apt-get install gcc make git g++ dnsmasq vim
+COPY ./api ./api
+COPY ./compute/scripts ./compute/scripts
+RUN pip install numpy
```
Comment (astaff) [inline Dockerfile.template]: Let's switch to `make install` Prior conversation: https://github.com/Opentrons/opentrons/pull/347#pullrequestreview-65177763 and https://github.com/Opentrons/opentrons/issues/370

File: api/Makefile [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 SHELL := /bin/bash
 
-.PHONY: test docs publish clean install exe dev
+.PHONY: test docs publish clean install exe dev info_device sync_device active_devices \
```
Comment (astaff) [inline api/Makefile]: Before we extend the list of make targets, as an MVP, I would suggest checking for `OT_DEVICE_ID` in `dev` target and possibly prompting to confirm that you indeed want to run on that device. Similar to app behavior, I want to see us running into collisions before we introduce a mechanism to resolve them.

File: api/opentrons/drivers/smoothie_drivers/v3_0_0/serial_communication.py [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 
 DRIVER_ACK = b'ok\r\nok\r\n'
 RECOVERY_TIMEOUT = 10
 DEFAULT_SERIAL_TIMEOUT = 5
 DEFAULT_WRITE_TIMEOUT = 30
-SMOOTHIE_BOARD_NAME = 'FT232R'
+
+SMOOTHIE_PORT_ID = 'FT232R'
+
+if os.environ.get('RUNNING_ON_PI'):
```
Comment (astaff) [inline api/opentrons/drivers/smoothie_drivers/v3_0_0/serial_communication.py]: Is RUNNING_ON_PI necessary?

'AMA' or anything else should be passed by whoever is instantiating us and the environment (OT_PORT_NAME or something) should be set in container.

File: compute/dnsmasq.conf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+# Set up a DHCP server on ethernet interface
+interface=eth0
+port=0
+dhcp-option=3 # disables default router
+dhcp-option=6 # disables DNS
+dhcp-range=192.168.42.2,192.168.42.254
```
Comment (astaff) [inline compute/dnsmasq.conf]: We are issuing an address in a class C private network which can coincide with the local network being used by the user. We should try to use link-local (https://trello.com/c/TNNimGV3/89-m-raspberry-pi-support-for-usb-connectivity)

**Comment by astaff** [comment]
Thanks for opening it for early feedback. Added a few comments that could be useful while you are preparing your PR for review. Didn't look into avahi / dbus work closely, assuming tests will be added to facilitate the review.

File: compute/dnsmasq.conf (hunk 4 above)
Comment (jaredgreene1) [reply compute/dnsmasq.conf]: Awesome point, thanks Artyom.

How defensive do you think we should be for this? Do you think selecting a pseudo-random static ip using the pi's mac might be enough (https://tools.ietf.org/html/rfc3927#section-2.1).

Should we perform collision detections on our static ip? Should we perform collision detections (https://tools.ietf.org/html/rfc3927#section-2.2)? Wouldn't be hard to use an arp probe to detect collisions, but do you think it's necessary?

**Pull request closed without merging**